src/main/scala/org/ergoplatform/local/MempoolAuditor.scala

```diff
@@ -0,0 +1,81 @@
+package org.ergoplatform.local
+
+import akka.actor.{Actor, ActorRef, ActorRefFactory, Props}
+import org.ergoplatform.modifiers.ErgoFullBlock
+import org.ergoplatform.modifiers.history.Header
+import org.ergoplatform.modifiers.mempool.ErgoTransaction
+import org.ergoplatform.nodeView.ErgoReadersHolder.{GetReaders, Readers}
+import org.ergoplatform.nodeView.mempool.ErgoMemPoolReader
+import org.ergoplatform.settings.NodeConfigurationSettings
+import scorex.core.NodeViewHolder.ReceivableMessages.EliminateTransactions
+import scorex.core.network.NodeViewSynchronizer.ReceivableMessages.SemanticallySuccessfulModifier
+import scorex.core.transaction.state.TransactionValidation
+import scorex.util.{ModifierId, ScorexLogging}
+
+import scala.annotation.tailrec
+import scala.util.Success
+
+/**
+  * Performs mempool validation in between blocks application.
+  */
+class MempoolAuditor(nodeViewHolderRef: ActorRef,
+                     readersHolderRef: ActorRef,
+                     nodeSettings: NodeConfigurationSettings) extends Actor with ScorexLogging {
+
+  override def preStart(): Unit = {
+    context.system.eventStream.subscribe(self, classOf[SemanticallySuccessfulModifier[_]])
+  }
+
+  override def receive: Receive = {
+    case SemanticallySuccessfulModifier(_: ErgoFullBlock) | SemanticallySuccessfulModifier(_: Header) =>
+      readersHolderRef ! GetReaders
+    case Readers(_, st: TransactionValidation[ErgoTransaction@unchecked], mp, _) =>
```

Comment:
Won't it fail in `Digest` mode? It is not `TransactionValidation`

Reply:
There is default branch for unsuitable readers